logstash-core/src/main/java/org/logstash/ackedqueue/PageFactory.java

```diff
@@ -0,0 +1,80 @@
+package org.logstash.ackedqueue;
+
+import org.logstash.ackedqueue.io.PageIO;
+
+import java.util.BitSet;
+
+public class PageFactory {
```

Comment:
@colinsurprenant slightly unrelated topic to this PR, but since we're going to expose our Java codebase eventually (hopefully soon :D), do we need to have this new class `public`? I think package private suffices here since only `Queue` uses it right?